### PR TITLE
TileSet/TileMap fixes

### LINKED ISSUE #25817: Atlas one way collision is not working
**Godot version:**
v3.1.beta3.official

**OS/device including version:**
macOS Mojave 10.14.3

**Issue description:**
Atlas one way collision is not working (it just works in the first tile collision), single tiles are working well.
Atlas tiles look unsorted, single tiles are hard to sort.

**Steps to reproduce:**
- Create a TileMap
- Create a TileSet
-- Add an Atlas
-- Create collisions
-- Turn on "Shape One Way" property

**Minimal reproduction project:**
[tileset_bug.zip](https://github.com/godotengine/godot/files/2855160/tileset_bug.zip)

### PR DESCRIPTION
**Pull request by MarianoGnu**
1. Decompose solid non-convex polygons into convexes. Real fix for #24003 
> Before:
> https://gyazo.com/78f161cccb23bf0a41b16d05e68db965
> After:
> https://gyazo.com/c940f20158f95f999f7dea4167021d03

2. TileSetEditor: Expose to the user OneWayCollision+Margin properties. Fixes #25817

### CODE REVIEW

**Comment by MarianoGnu** [comment]
@reduz please review this.

File: core/math/geometry.h [hunk 1]
```diff
@@ -950,6 +951,40 @@ class Geometry {
 		return H;
 	}
 
+	static Vector<Vector<Vector2> > decompose_polygon_in_convex(Vector<Point2> polygon) {
```
Comment (akien-mga) [inline core/math/geometry.h]: Why does this have to be in the header instead of the .cpp?

Comment (akien-mga) [reply core/math/geometry.h]: It makes build fail BTW:
```
clang++ -o platform/x11/os_x11.x11.opt.64.llvm.o -c -O3 -pipe -Wall -Werror -DZSTD_STATIC_LINKING_ONLY -DFREETYPE_ENABLED -DSVG_ENABLED -DTYPED_METHOD_BIND -DTOUCH_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DJOYDEV_ENABLED -DX11_ENABLED -DUNIX_ENABLED -DOPENGL_ENABLED -DGLES_ENABLED -DGLAD_ENABLED -DGLES_OVER_GL -DNDEBUG -DPTRCALL_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DHAVE_MNTENT -Ieditor -I. -I/usr/include/freetype2 -I/usr/include/libpng12 -Iplatform/x11 -Ithirdparty/zstd -Ithirdparty/glad -Ithirdparty/nanosvg platform/x11/os_x11.cpp
In file included from platform/x11/godot_x11.cpp:37:
In file included from platform/x11/os_x11.h:45:
In file included from ./servers/visual/rasterizer.h:35:
In file included from ./servers/visual_server.h:36:
In file included from ./core/math/geometry.h:42:
./thirdparty/misc/triangulator.h:180:7: error: expected member name or ';' after
      declaration specifiers
        bool Below(Vector2 &p1, Vector2 &p2);
        ~~~~ ^
/usr/include/X11/X.h:463:33: note: expanded from macro 'Below'
#define Below                   1
                                ^
```

**Comment by MarianoGnu** [comment]
Talking with @reduz he mentioned that he wants to solve #24003 with a different aproach, so for now i'll close this PR and open a new one just for #25817.

**Comment by akien-mga** [comment]
Superseded by #26614 and #26599.

**Pull request closed without merging**